opm/input/eclipse/EclipseState/Grid/Carfin.cpp

```diff
@@ -179,6 +181,44 @@ namespace Opm
         return m_dims[0] * m_dims[1] * m_dims[2];
     }
 
+    std::size_t Carfin::num_parent_cells() const
+    {
+        return (upper(0) - lower(0) + 1) * 
+               (upper(1) - lower(1) + 1) *
+               (upper(2) - lower(2) + 1);
+    }
+
+    std::tuple<std::vector<std::size_t>, 
+               std::vector<std::size_t>, 
+               std::vector<std::size_t>>
+    Carfin::parent_cellsIJK() const
+    {
+        std::vector<std::size_t> i_list;
+        std::vector<std::size_t> j_list;  
+        std::vector<std::size_t> k_list;
+        std::size_t list_size =  num_parent_cells();
+        
+        i_list.resize(list_size);
+        j_list.resize(list_size);
+        k_list.resize(list_size);
+        std::size_t index = 0;
+        for (std::size_t i_index = lower(0); i_index <= upper(0); i_index++)
```

Comment:
> Prefer pre-increment in for-loops?

On another note, the **typical** loop order for traversing cells is I :arrow_right: J :arrow_right: K&ndash;meaning I looping faster than J and J looping faster than K, occasionally referred to as "natural ordering"&ndash;so unless there's some kind external requirement that we use K :arrow_right: J :arrow_right: I here, I'd prefer that the loop order be reversed to
```
for (k_index = ...) {
    for (j_index = ...) {
        for (i_index = ...) {
            ...
        }
    }
}
```
That said, this doesn't **really** need to be a member function at all, since client code can use the `CARFIN` [index bounds](https://github.com/OPM/opm-common/blob/5c919d2d9d6f150b6a85ea9f9a5808a44595e043/opm/input/eclipse/EclipseState/Grid/Carfin.hpp#L88-L93) directly without forming "large" vectors of index tuples.  In the use case of establishing the global cell IDs you could write something like
```C++
for (auto k = lgr.K1(); k <= lgr.K2(); ++k) {
    for (auto j = lgr.J1(); j <= lgr.J2(); ++j) {
        for (auto i = lgr.I1(); i <= lgr.I2(); ++i) {
            globCell.push_back(this->activeIndex(i, j, k));
        }
    }
}
```